### DISCUSSION TITLE
$watch doesn't return the old value

### DISCUSSION THREAD

**Discussion by Babelfisch** [alpinejs/alpine#3467] [7. Bugs (archived)]
I tried this simple code of the `$watch` documentation for deep watching:

```js
<div x-data="{ foo: { bar: 'baz' }}" x-init="$watch('foo', (value, oldValue) => console.log(value, oldValue))">
    <button @click="foo.bar = 'bob'">Update</button>
</div>
```

The expected output is:

```
{bar: "bob"}
{bar: "baz"}
```

But the real output is:
```
{bar: "bob"}
{bar: "bob"}
```

Tested with Safari and Chrome and AlpineJS 3.12.0 and 3.11.1

BTW: Why it's not possible to open a issue?

**Comment by ekwoka** [comment]
Alpine has issues closed. Only discussions.

Technically, your code is working, and it is returning the old value.

The old value is also the same object as the new value.

**Reply by SimoTod** [reply]
I believe there's an "on hold" PR to fix it that CaleB will review after livewire v3 is released

**Reply by anjanesh** [reply]
Why is the old value the same as the new value ?
I would like it to return the old value as `{ foo: { bar: 'baz' }}`

**Reply by ekwoka** [reply]
Because the old value and the new value are the same thing.

```js
const objA = {
  Foo: 'bar'
}

const objB = objA

objA.Foo = 'baz'

objA.Foo === objB.Foo
```

It's just how JS works.

**Reply by KRHugo** [reply]
I understand that it's how javascript works.

However, the document explains that the previous value will be different.

https://alpinejs.dev/magics/watch#deep-watching

**Reply by ekwoka** [reply]
Ooh, interesting then it's an unintended bug.

**Comment by joshhanley** [comment]
@Babelfisch we are now using discussions to track bugs and ideas instead of issues. See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423

**Comment by Fak3** [comment]
I am facing the same issue. Is there any known workaround? How can i get the old value of the json object?

**Comment by whitefang57** [comment]
Also facing this issue as well! I'd appreciate if that limitation could be better documented in the [deep-watching](https://alpinejs.dev/magics/watch#deep-watching) section of the docs, or if someone could take a look at #4406.
